Global.h

```diff
@@ -698,3 +698,16 @@ namespace vstd
 }
 using vstd::operator-=;
 using vstd::make_unique;
+
+#if defined VCMI_ANDROID && defined NO_STD_TOSTRING

```

Comment:
I don't exactly know how that works. Generally there are several runtimes that can be used to build native code (see: https://developer.android.com/ndk/guides/cpp-support.html#hr ). The last time I tried to experiment with these, the full libc++ didn't work correctly (compiled, but there were errors during runtime). I don't know why gnustl (which is, according to the page I linked "This runtime is the GNU Standard C++ Library, (libstdc++-v3)") doesn't have that method -- it seems it is possible to circumvent the missing methods in some way (example here http://stackoverflow.com/questions/17950814/how-to-use-stdstoul-and-stdstoull-in-android/18124627#18124627 ), but I didn't try that.